pkg/ccl/sqlproxyccl/server.go

```diff
@@ -162,10 +163,24 @@ func (s *Server) handleCancel(w http.ResponseWriter, r *http.Request) {
 // a health check endpoint at /_status/healthz, and pprof debug
 // endpoints at /debug/pprof.
 func (s *Server) ServeHTTP(ctx context.Context, ln net.Listener) error {
-	srv := http.Server{
-		Handler: s.mux,
+	if s.handler.RequireProxyProtocol {
```

Comment:
I know I'm the one who initially said we probably needed this, but I'm wondering if that's true. For HTTP traffic the load balancer will usually add some kind of HTTP header to tell you the original IP address, so I kinda doubt we would ever need this?

Reply:
My understanding with NLBs is that once we enable the proxy protocol, healthchecks will use that as well: https://docs.aws.amazon.com/elasticloadbalancing/latest/network/load-balancer-target-groups.html#proxy-protocol. I know that we’re planning to use TCP instead of HTTP, but I wanted this here for flexibility, in case we decide to use HTTP in the future.

Reply:
> After you enable proxy protocol, the proxy protocol header is also included in health check connections from the load balancer. However, with health check connections, the client connection information is not sent in the proxy protocol header.

Reply:
Yes, that is possible, but you need to update the network policies to allow access to the HTTP port. I’m leaning towards leaving the change as-is, as it gives us more flexibility in the future.

Reply:
I'm okay with leaving this, we just need to not trust the IP address from the HTTP proxy header unless we change it to be required.